core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -683,9 +684,14 @@ private[sql] final case class ColumnTableScan(
             """.stripMargin
         dictionaryAssignCode =
             s"$dictIndex = $decoder.readDictionaryIndex($buffer, $cursorVar);"
+        val nullCheckAddon =
+          if (notNullVar != null) s"if ($notNullVar < 0) $nullVar = $col == null;\n"
+          else ""
```

Comment:
This will leave an empty semi-colon since its both here and in DictionaryOptimizedAccessor. I guess clearer for code clarity vs in generated code.